packages/lesswrong/components/votes/NamesAttachedReactionsVoteOnComment.tsx

```diff
@@ -0,0 +1,630 @@
+  reactionCount: {
+    fontSize: 14,
+    fontFamily: theme.typography.commentStyle.fontFamily,
+    color: theme.palette.text.dim60,
+    marginLeft: 3,
+    verticalAlign: "middle",
+  },
+  addReactionButton: {
+    verticalAlign: "bottom",
+    marginLeft: 8,
+    filter: "opacity(0.4)",
+    "& svg": {
+      width: 18,
+      height: 18,
+    },
+    "&:hover": {
+      filter: "opacity(0.8)",
+    },
+  },
+  reactOrAntireact: {
+    marginLeft: 12,
+  },
+  hoverBallot: {
+    fontFamily: theme.typography.commentStyle.fontFamily,
+    paddingTop: 12,
+    paddingBottom: 12,
+    maxWidth: 350,
+  },
+  hoverBallotEntry: {
+    fontFamily: theme.typography.commentStyle.fontFamily,
+    cursor: "pointer",
+    paddingTop: 4,
+    paddingBottom: 4,
+    paddingLeft: 16,
+    paddingRight: 16,
+    "&:hover": {
+      background: theme.palette.panelBackground.darken04,
+    },
+  },
+  hoverBallotLabel: {
+    verticalAlign: "middle",
+    marginLeft: 6,
+    display: "inline-block",
+    minWidth: 80,
+  },
+  hoverBallotReactDescription: {
+    marginLeft: 25,
+    marginBottom: 6,
+    fontSize: 11,
+  },
+  usersWhoReacted: {
+    marginLeft: 25,
+    fontSize: 11,
+  },
+  alreadyUsedReactions: {
+    marginBottom: 12,
+  },
+
+  reactionVoteCount: {
+    display: "inline-block",
+    verticalAlign: "middle",
+  },
+  voteArrow: {
+    color: theme.palette.grey[400],
+  },
+  voteArrowIcon: {
+    fontSize: 'inherit',
+    padding: 0,
+    width: 24,
+    height: 24,
+    verticalAlign: "middle",
+  },
+  voteArrowLeft: {
+    transform: 'rotate(-90deg)',
+    marginRight: -4,
+  },
+  voteArrowRight: {
+    transform: 'rotate(-270deg)',
+    marginLeft: -4,
+  },
+})
+
+
+const useNamesAttachedReactionsVoting = (voteProps: VotingProps<VoteableTypeClient>): {
+  currentUserExtendedVote: NamesAttachedReactionsVote|null,
+  getCurrentUserReactionVote: (name: string) => VoteOnReactionType|null,
+  toggleReaction: (name: string) => void
+  setCurrentUserReaction: (name: string, reaction: VoteOnReactionType|null) => void,
+}=> {
+  const { openDialog } = useDialog()
+  const currentUser = useCurrentUser()
+  const currentUserExtendedVote = (voteProps.document?.currentUserExtendedVote as NamesAttachedReactionsVote|undefined) ?? null;
+
+  function getCurrentUserReactionVote(name: string): VoteOnReactionType|null {
+    const reacts = currentUserExtendedVote?.reacts ?? [];
+    const relevantVoteIndex = reacts.findIndex(r=>r.react===name);
+    if (relevantVoteIndex < 0) return null;
+    return reacts[relevantVoteIndex].vote;
+  }
+
+  function openLoginDialog() {
+    openDialog({
+      componentName: "LoginPopup",
+      componentProps: {}
+    })
+  }
+
+  function toggleReaction(name: string) {
+    if (!currentUser) {
+      openLoginDialog();
+      return;
+    }
+    
+    if (getCurrentUserReactionVote(name)) {
+      clearCurrentUserReaction(name);
+    } else {
+      const initialVote = "created"; //TODO: "created" vs "seconded"
```

Comment:
This has a TODO - not sure if still relevant?

Reply:
Tracking who created vs who seconded is a nice-to-have but not essential (particularly not for the beta-test open thread). So, leaving in the TODO for now.